packages/tradershub/src/features/cfd/modals/JurisdictionModal/JurisdictionModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useAvailableMT5Accounts } from '@deriv/api';
+import { Provider } from '@deriv/library';
+import { Button, Heading, useBreakpoint } from '@deriv/quill-design';
+import { Modal } from '../../../../components/Modal';
+import { DummyComponent } from '../../components/DummyComponent';
+import { DynamicLeverageContext } from '../../components/DynamicLeverageContext';
+import { Jurisdiction, MarketType, MarketTypeDetails } from '../../constants';
+import { DynamicLeverageScreen, DynamicLeverageTitle } from '../../screens/DynamicLeverage';
+import { JurisdictionScreen } from '../../screens/Jurisdiction';
+
+const JurisdictionModal = () => {
+    const [selectedJurisdiction, setSelectedJurisdiction] = useState('');
+    const [isDynamicLeverageVisible, setIsDynamicLeverageVisible] = useState(false);
+    const [isCheckBoxChecked, setIsCheckBoxChecked] = useState(false);
+
+    const { show } = Provider.useModal();
+    const { getCFDState, setCfdState } = Provider.useCFDContext();
+
+    const { isLoading } = useAvailableMT5Accounts();
+    const { isMobile } = useBreakpoint();
+
+    const marketType = getCFDState('marketType') ?? MarketType.ALL;
+
+    const { title } = MarketTypeDetails[marketType];
+
+    const toggleDynamicLeverage = useCallback(() => {
+        setIsDynamicLeverageVisible(!isDynamicLeverageVisible);
+    }, [isDynamicLeverageVisible, setIsDynamicLeverageVisible]);
+
+    const jurisdictionTitle = `Choose a jurisdiction for your Deriv MT5 ${title} account`;
+
+    const JurisdictionFlow = () => {
+        if (selectedJurisdiction === Jurisdiction.SVG) {
+            return <DummyComponent />; // MT5PasswordModal
+        }
+
+        return <DummyComponent />; // Verification flow
+    };
+
+    useEffect(() => {
+        setCfdState('selectedJurisdiction', selectedJurisdiction);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we shouldn't suppress this warning unless we're very sure

Reply:
Warning suggests that we should add setCfdState to the dependency array, but adding it to dependency array i noticed that it's continuously re-rendering.

Reply:
in that case `setCfdState` is being reinitialized on each render. We should wrap setCfdState in a callback in the context and remove this disable rule
it'll solve the issue for future uses as well

Reply:
setCfdState is already wrapped in useCallback but the dependency array in that useCallback is cfdState itself which I am updating here.